---
title: "metaG_Beta"
author: "SRomano"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Load packages

```{r setup}
library(phyloseq)
library(dplyr)
library(microbiome)
library(vegan)
library(RColorBrewer)
library(cowplot)

set.seed(435)
```

# Load metadata + datasets
```{r}
meta<-readRDS("/metaG/R_analyses/Taxonomy/RDS/meta.rds")
all.g<-readRDS("/metaG/R_analyses/Taxonomy/RDS/all_no0_TSS_noUnassigned.rds")
```

Check if in the meta SapleID match with the one in the mOTUs table.
Add info about geography.

```{r}
stopifnot(all(meta$SampleID == names(all.g)))
message("Are all samples in meta identical to the microbiome table? ", all(meta$SampleID == names(all.g)))

# add countries and continent w/e

w<-c("Bedarf_2017","Boktor_1_2023","Boktor_2_2023", "Wallen_2022")
e<-c( "Mao_2021", "Qian_2020","Jo_2022")
meta$WE<-ifelse(meta$Study %in% w, "West", "Est")
meta$Continent<-ifelse(meta$Study %in% e, "Asia", "America")
meta$Continent<-ifelse(meta$Study == "Bedarf_2017", "Europe", meta$Continent)
meta$Country<-ifelse(meta$Study == "Bedarf_2017", "Germany", "China")
meta$Country<-ifelse(meta$Study == "Jo_2022", "Korea", meta$Country)
meta$Country<-ifelse(meta$Study  %in% c("Boktor_1_2023","Boktor_2_2023", "Wallen_2022"), "USA", meta$Country)

```

# Beta
Investigate BC dissimilarities between vs within
```{r, fig.width = 10, fig.height = 10}
rownames(meta)<-meta$SampleID
all_phylo<-phyloseq(otu_table(all.g,taxa_are_rows = T), sample_data(meta))

all_bc<-phyloseq::distance(all_phylo, method = "bray", square = T)

#####################################
# Compare within vs between distances
#####################################
source("/Scripts/compare.groups.r")
l.sm <- distance.between.groups(dist.matrix = all_bc,
                                meta =  meta, 
                                variable.study = "Study",
                                same = T)
l.diff <- distance.between.groups(dist.matrix = all_bc,
                                meta =  meta, 
                                variable.study = "Study",
                                same = F)
sm<-l.sm %>%
  unlist(recursive = F) %>%
  as.vector()
bet<-l.diff %>%
  unlist(recursive = F) %>%
  as.vector()
df.compare<-data.frame(dist = c(sm,bet),
                       wb = c(rep("Within Studies", length =  length(sm)),
                              rep("Between Studies", length = length(bet))))

# t-test + Cohen's D effect size
t.test(df.compare$dist ~ df.compare$wb, var.equal = T)
(cd<-rstatix::cohens_d(data = df.compare,
                         formula = dist~wb, 
                         paired = F, var.equal = T, ci =T, nboot = 333))

df.compare$wb<-factor(df.compare$wb, 
                      levels = c("Within Studies", "Between Studies"), 
                      ordered = T)

df.compare$seq<-rep("SMG", nrow(df.compare))
saveRDS(df.compare, "/metaG/R_analyses/Taxonomy/RDS/df.compare.beta.rds")
```


## Permanova

```{r}
# define the permutation schema
perm<-how(blocks = as.factor(meta$Study), nperm = 2000)

## Desease
(a.pd.bc<-adonis2(all_bc ~ PD, data = microbiome::meta(all_phylo), permutations = perm))

## Study
(a.st.bc<-adonis2(all_bc ~ Study, data = microbiome::meta(all_phylo), permutations = 2000))

## continent
(a.cont.bc<-adonis2(all_bc ~ Continent, data = microbiome::meta(all_phylo), permutations = 2000))

## country
(a.count.bc<-adonis2(all_bc ~ Country, data = microbiome::meta(all_phylo), permutations = 2000))

## we
(a.count.bc<-adonis2(all_bc ~ WE, data = microbiome::meta(all_phylo), permutations = 2000))

```

## Ordination conditioned by study
```{r}
nb.cols <- 3
mycolors <- colorRampPalette(brewer.pal(8, "Set2"))(nb.cols)
 
all_dbrda_bc<-ordinate(all_phylo, method = "CAP", distance = all_bc, formula = all_phylo ~ 1  + Condition(Study))
plot_ordination(all_phylo, all_dbrda_bc, "samples", shape="PD", color = "Study") +
                theme_bw() +
                scale_fill_manual(values = mycolors) +
                geom_point(size = 2.5) +
                labs(color = "Study", shape = "Status") +
                theme(legend.title = element_text(size = 12, face = "bold"),
                      legend.text = element_text(size = 12)) +
    ggtitle("dbRDA-Conditioned Study - BC")

# Export ordination data fro plotting with 16S
sc<-scores(all_dbrda_bc, display = "sites") %>% as.data.frame()
meta<-meta[order(meta$SampleID),]
all(rownames(sc) == meta$SampleID)

sc.m<-cbind.data.frame(sc, meta)
saveRDS(sc.m, "/metaG/R_analyses/Taxonomy/RDS/dbRDA_df.rds")


```

# Ordination not conditioned
```{r}
all_dbrda_bc<-ordinate(all_phylo, method = "CAP", distance = all_bc, formula = all_phylo ~ 1)

# Quick plot
plot_ordination(all_phylo, all_dbrda_bc, "samples", shape="PD", color = "Study") +
                theme_bw() +
                scale_fill_manual(values = mycolors) +
                geom_point(size = 2.5) +
                labs(color = "Study", shape = "Status") +
                theme(legend.title = element_text(size = 12, face = "bold"),
                      legend.text = element_text(size = 12)) +
    ggtitle("dbRDA NOT-Conditioned Study - BC")


# Export ordination data fro plotting with 16S
sc<-scores(all_dbrda_bc, display = "sites") %>% as.data.frame() # extract the coordinates
meta<-meta[order(meta$SampleID),]
all(rownames(sc) == meta$SampleID)

sc.m<-cbind.data.frame(sc, meta)
saveRDS(sc.m, "/metaG/R_analyses/Taxonomy/RDS/dbRDA_df_study.rds")
```
Save workspace
```{r}
d<-"/metaG//R_analyses/Taxonomy/RData/"
save.image(paste0(d, "metaG_beta.RData"))
```
