---
title: "filter"
author: "SRomano"
output: html_document
editor_options: 
  chunk_output_type: console
---

Load the required packages.

```{r setup, include=FALSE}
list.of.packages <- c("dplyr")
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])]
if(length(new.packages)) install.packages(new.packages)
if (!requireNamespace("BiocManager", quietly = TRUE)){
    install.packages("BiocManager")}


list.of.bioc <- c("phyloseq", "microbiome")
new.packages <- list.of.bioc[!(list.of.bioc %in% installed.packages()[,"Package"])]
if(length(new.packages)) BiocManager::install(new.packages)
library(dplyr)
library(microbiome)

source("/Scripts/Pevalence_functions.r")
d<-"/16S/R_analyses/RData/"
dir.create(d, recursive = T)

```

# Load metadata + datasets
```{r}
meta<-readRDS("/16S/R_analyses/RDS/meta.rds")
all.g<-readRDS("/all.g_combined_NB_GTDB207.rds")
```
# Clen data
```{r}
dim(meta)
dim(all.g)

# are all samples in meta in the OTU table?
meta<-meta[order(meta$SampleID),]
all.g<-all.g[, order(names(all.g))]
if((all(meta$SampleID == names(all.g)))){
  print("All SampleID in meta are the same as in the Abund table")
} else {stop("meta and Abund table do not have the same samples")}

```
# Clean the datasets

```{r}
if(length(which(rowSums(all.g)==0)) > 0){
  print("Removing taxa with 0 counts, most likely due to sample filtration")
  message(paste0("There are ", length(which(rowSums(all.g)==0)), " taxa with 0 counts. removed!"))
  all.g<-all.g[-1*which(rowSums(all.g)==0),]
} else {print("No taxa with 0 counts")}

if(length(all.g[, colSums(all.g)==0])  > 0){
  print("Removing samples with 0 counts, most likely due to sample filtration")
  message(paste0("There are ",length(which(colSums(all.g)==0)), " samples with 0 counts. removed!"))
  all.g<-all.g[,-1*which(colSums(all.g)==0)]
} else {print("No samples with 0 counts")}

```

## Remove samples with less than 2000 reads
```{r}
colSums(all.g) -> cls.g
df.g<-data.frame(samples = names(cls.g),
                 depth = cls.g)


all(meta$SampleID == names(all.g))
df.g<-cbind.data.frame(df.g, meta)


df.g<-df.g[order(df.g$depth),]
head(df.g, n = 20)

ids<-subset(df.g, depth >= 2000)$SampleID
###############
# Filter both ASVs and meta
############

all.g<-all.g[, ids]
meta<-meta[meta$SampleID %in% ids,]
meta<-meta[order(meta$SampleID),]
all.g<-all.g[, order(names(all.g))]
all(names(all.g) == meta$SampleID)
saveRDS(meta, "/16S/R_analyses/RDS/meta.rds")
dim(meta)
```

Now save the files
```{r}
print("Save a count table and a proportion table")
all.g.count<-all.g
all.g<-as.data.frame(prop.table(as.matrix(all.g.count), 2))

all.g<-all.g[,order(names(all.g))]
meta<-meta[order(meta$SampleID),]
if(all(meta$SampleID == names(all.g)) & all(meta$SampleID == names(all.g.count))){
  saveRDS(all.g.count, "/16S/R_analyses/RDS/all_g_counts.rds")
  saveRDS(all.g, "/16S/R_analyses/RDS/all_g_TSS.rds")
}
dim(all.g)

```

# Detection filtration
## Load list_df
```{r}
all.g<-readRDS("/List_datasets_NB_GTDB207_25Aug2022.rds")

print("Convert the data into numeric")
for(i in 1:length(all.g)){
  tax<-row.names(all.g[[i]])
  all.g[[i]]<-apply(all.g[[i]], as.numeric, MARGIN = 2) %>% as.data.frame()
  row.names(all.g[[i]])<-tax
}

print("Remove taxa with 0 and samples with 0")
for(i in 1:length(all.g)){
  all.g[[i]]<-all.g[[i]][, names(-all.g[[i]]) %in% ids] # select only samples with 2000 reads

  message(paste0("There are ", length(which(rowSums(all.g[[i]])==0)), " taxa with 0 counts. removed!"))
  all.g[[i]]<-all.g[[i]][which(rowSums(all.g[[i]])!=0),]
}


print("Remove samples with 0 and samples with 0")
for(i in 1:length(all.g)){
  message(paste0("There are ",length(which(colSums(all.g[[i]])==0)), " samples with 0 counts. removed!"))
  all.g[[i]]<-all.g[[i]][,which(colSums(all.g[[i]])!=0)]

}


print("Transform in prop")
all.g.prop<-all.g
for(i in 1:length(all.g)){
  all.g.prop[[i]]<-as.data.frame(prop.table(as.matrix(all.g[[i]]), margin = 2))
}

x<-unlist(lapply(all.g, function(x) names(x)))
y<-unlist(lapply(all.g.prop, function(x) names(x)))
all(x==y)

if(all(x[order(x)] == meta$SampleID)){
  print("all Abund table have the same samples than meta")
  saveRDS(all.g, "/16S/R_analyses/RDS/l.df_g_counts.rds")
  saveRDS(all.g.prop, "/16S/R_analyses/RDS/l.df_g_TSS.rds")
} else {message("Samples in meta and abund table do not match!")}



```

# Calculate and save filtration
```{r}
l_df<-readRDS("/16S/R_analyses/RDS/l.df_g_TSS.rds")

spec_5x10<-prevalence.matching(l_df, n.studies = 10, prev = 5) 
saveRDS(spec_5x10, "/16S/R_analyses/RDS/spec_5x10.rds")

spec_10x10<-prevalence.matching(l_df, n.studies = 10, prev = 10)
saveRDS(spec_10x10, "/16S/R_analyses/RDS/spec_10x10.rds")

spec_20x10<-prevalence.matching(l_df, n.studies = 10, prev = 20)
saveRDS(spec_20x10, "/16S/R_analyses/RDS/spec_20x10.rds")

spec_30x10<-prevalence.matching(l_df, n.studies = 10, prev = 30)
saveRDS(spec_30x10, "/16S/R_analyses/RDS/spec_30x10.rds")
```
